SimformSolutionsPvtLtd/Android_Project_Setup
# Update Internet Connectivity Code & Update Libs

### **Update Internet Connectivity Code**

**Tested on:**

1. Samsung M12 - Android 11
2. Motorola g51 - Android 12
3. Emulator with API levels 32 and 33

**Test Cases:**

1. Without connecting to any network
    1. Open the app → It should receive a callback for no network-connected
    2. Turn on the mobile data or wifi → It should receive a callback for the connected state
    3. Turn off either mobile data or wifi  → It should stay connected
    4. Turn off both mobile data and wifi →  It should receive a callback for a disconnected state
2. With connecting to any network
    1. Open the app →  It should receive a callback for the connected state
    2. Turn off either mobile data or wifi  → It should stay connected
    3. Turn off both mobile data and wifi →  It should receive a callback for a disconnected state
3. Internet availability while connected to any network
    1. Open the app with no internet -> It should receive a callback for a disconnected
    2. Internet unavailable  → It should receive a callback for disconnected (Once an exclamation mark appears on the connected network)
    3. Internet available → It should receive a callback for connection (Once the exclamation mark disappears on the connected network)
    
**Videos:**

[Video 1](https://simformsolutionspvtltd-my.sharepoint.com/:v:/r/personal/jaydeep_simformsolutions_com/Documents/Internet%20Connectivity/Video%201.mp4?csf=1&web=1&nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=UYd9e5)
[Video 2](https://simformsolutionspvtltd-my.sharepoint.com/:v:/r/personal/jaydeep_simformsolutions_com/Documents/Internet%20Connectivity/Video%202.mp4?csf=1&web=1&nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=dHv2Px)
[Video 3](https://simformsolutionspvtltd-my.sharepoint.com/:v:/g/personal/jaydeep_simformsolutions_com/ESJ5FPqoVNtMolOhfey51G8BWb0FrCPPQ4V5DdEbvTVr-w?nav=eyJyZWZlcnJhbEluZm8iOnsicmVmZXJyYWxBcHAiOiJPbmVEcml2ZUZvckJ1c2luZXNzIiwicmVmZXJyYWxBcHBQbGF0Zm9ybSI6IldlYiIsInJlZmVycmFsTW9kZSI6InZpZXciLCJyZWZlcnJhbFZpZXciOiJNeUZpbGVzTGlua0NvcHkifX0&e=87s1dR)
   

### **Update Libraries**
